### PR TITLE
[FilesCollection] Allow getFiles with dot syntax

### LINKED ISSUE #112: [FilesCollection] Allow getFiles with dot syntax
We can currently check if a file exists using dot syntax to dig into arrays, but need to implement this feature for `getFile()` also.

Enhancement Checklist:
- [ ] Component(s) modified
- [ ] Unit testing updated
- [ ] User guide updated
- [ ] Securely signed commits
- [ ] Conforms to style guide

### PR DESCRIPTION
**Pull request by davidgv88**
[FilesCollection] Allow getFiles with dot syntax. Fix #112

### CODE REVIEW

File: user_guide_src/source/libraries/uploaded_files.rst [hunk 1]
```diff
@@ -69,9 +69,7 @@ In this case, the returned array of files would be more like::
 
 If you just need to access a single file, you can use ``getFile()`` to retrieve the file instance directly::
 
-	$file = $this->request->getFile('avatar');
-
-.. note:: This currently only works with simple file names and not with array syntax naming.
+	$file = $this->request->getFile('my-form.details.avatar.0');
```
Comment (lonnieezell) [inline user_guide_src/source/libraries/uploaded_files.rst]: A couple of things: 

1. We should still show that you can access the simple case `$this->request->getFile('avatar');`. 
2. If there are multiple files with the same name, then your example applies, but what if there's only one file with a deep name? Basically - we should show all three methods in the docs. 

```
// Get a single file
$file = $this->request->getFile('avatar');
// Get a single file within a more complex path
$file = $this->request->getFile('my-form.details.avatar');
// Get the first of several files uploaded under the same name
$file = $this->request->getFile('my-form.details.avatar.0');
```

Looking at the code, I believe those cases will all pass currently, and we need them to. :)

Comment (davidgv88) [inline user_guide_src/source/libraries/uploaded_files.rst]: Hi @lonnieezell Thanks!

Sorry for the brevity in the User Guide. My English is very poor.  In the next few days I will write some examples

**Comment by davidgv88** [comment]
Hi @lonnieezell 

Done! I have written some examples. If necessary can you correct the user guide?

Thanks!

**Comment by lonnieezell** [comment]
Looks good. Thanks! Merging.

**Pull request merged**